### ISSUE TITLE
Invites don't show the name or alias of the room you are being invited to.

### ISSUE THREAD

**Issue by ara4n** [vector-im/vector-web#321]
Instead they show only the name of the person who has invited you. In the main room list, I suggest we show the name of the room as it would appear in Recents.  On the 'join' screen we would then be told who actually invited you.

**Comment by Kegsay** [comment]
Fixed in https://github.com/matrix-org/matrix-react-sdk/commit/8edfd60b643ac952447cb5e6ed31cee270825dd8

**Comment by ara4n** [comment]
This is still broken - @matthew:matrix.org sent an invite to @matthew2:matrix.org to join #test:matrix.org (via '/invite @matthew2:matrix.org' from Vector whilst in the room).  The invite was shown in matthew2's vector as "Invite from @matthew:matrix.org" rather than "Invite to #test:matrix.org". Separately, the push notification that matthew2 received leaked an ugly room ID.

**Comment by ara4n** [comment]
This looks to be a serverside issue - we simply don't receive the data from synapse :(

@NegativeMjark - please can you get this on your radar or bounce it to @erikjohnston or whoever got it wrong first time.

    {
      "start": "s8203666_18723_1633_237521_2134",
      "chunk": [
        {
          "prev_content": {
            "membership": "leave"
          },
          "origin_server_ts": 1450020720509,
          "sender": "@matthew:matrix.org",
          "event_id": "$145002072025781nccIP:matrix.org",
          "age": 220,
          "unsigned": {
            "prev_content": {
              "membership": "leave"
            },
            "prev_sender": "@matthew2:matrix.org",
            "age": 220,
            "invite_room_state": [
              {
                "content": {
                  "join_rule": "public"
                },
                "type": "m.room.join_rules",
                "sender": "@matthew:matrix.org",
                "state_key": ""
              }
            ],
            "replaces_state": "$145002009025089tPIvz:matrix.org"
          },
          "state_key": "@matthew2:matrix.org",
          "content": {
            "membership": "invite"
          },
          "room_id": "!vfFxDRtZSSdspfTSEr:matrix.org",
          "invite_room_state": [
            {
              "content": {
                "join_rule": "public"
              },
              "type": "m.room.join_rules",
              "sender": "@matthew:matrix.org",
              "state_key": ""
            }
          ],
          "user_id": "@matthew:matrix.org",
          "replaces_state": "$145002009025089tPIvz:matrix.org",
          "type": "m.room.member"
        }
      ],
      "end": "s8203695_18723_1633_237521_2134"
    }

**Comment by Kegsay** [comment]
According to the spec, the following event types are passed through:
 - "m.room.join_rules", "m.room.canonical_alias", "m.room.avatar", "m.room.name"

`#test:matrix.org` was created well before most of these things were A Thing (specifically `m.room.canonical_alias`) so there would be no alias to send to the client. Indeed, checking that room on matrix.org/beta does not show any canonical alias for that room.

**Comment by ara4n** [comment]
So basically this is a server bug to say "please provide a random alias if there's no canonical one"?

**Comment by Kegsay** [comment]
I don't think the server should lie and fabricate a fake `m.room.canonical_alias` event. However, it may be acceptable for `m.room.aliases` to be included in the allowed event types and leave it to clients to do a fallback display.

The general case will never produce what is happening to `#test:matrix.org` because the nominal flow is to create a room with a given alias (and `/createRoom` will automatically set the `canonical_alias` based on this). We don't actually expose any way in Vector to add or remove aliases from a room, so this is only going to be an issue for existing rooms.

The reason why `#test:matrix.org` is shown as that *anyway* is because of the fallback logic existing in the JS SDK to resort to using `m.room.aliases` rather than using the room ID. Using the same logic here feels acceptable, but has all the pitfalls associated with alias use (specifically that two clients may pick different aliases).

tl;dr - Add `m.room.aliases` to the `invite_room_state` and everything works.

**Comment by Kegsay** [comment]
I don't believe this is a vector issue any longer. Will leave this for a bit then close.

**Comment by ara4n** [comment]
ok. will raise as syn issue

**Comment by ara4n** [comment]
Raised as SYN-551
